router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -0,0 +1,325 @@
+	resolveURL := func(base, relative string) string {
+		// var logger logger.Logger
+		baseURL, _ := url.Parse(base)
+		// if err != nil {
+		// 	logger.Fatal(err)
+		// }
+		relURL, _ := url.Parse(relative)
+		// if err != nil {
+		// 	logger.Fatal(err)
+		// }
+		destURL := baseURL.ResolveReference(relURL).String()
+		return destURL
+	}
+	destinationID := destination.ID
+	destinationUploadUrl := asyncDestStruct.URL
+	url := resolveURL(b.TransformUrl, destinationUploadUrl)
+	filePath := asyncDestStruct.FileName
+	destConfig := destination.Config
+	destType := destination.DestinationDefinition.Name
+	failedJobIDs := asyncDestStruct.FailedJobIDs
+	importingJobIDs := asyncDestStruct.ImportingJobIDs
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		panic("BRT: Read File Failed" + err.Error())
+	}
+	defer file.Close()
+	var input []common.AsyncJob
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var tempJob common.AsyncJob
+		jobBytes := scanner.Bytes()
+		err := json.Unmarshal(jobBytes, &tempJob)
+		if err != nil {
+			panic("Unmarshalling a Single Line Failed")
+		}
+		input = append(input, tempJob)
+	}
+	payload, err := json.Marshal(common.AsyncUploadT{
+		Input:    input,
+		Config:   destConfig,
+		DestType: strings.ToLower(destType),
+	})
+	if err != nil {
+		panic("BRT: JSON Marshal Failed " + err.Error())
+	}
+
+	uploadTimeStat := stats.Default.NewTaggedStat("async_upload_time", stats.TimerType, map[string]string{
+		"module":   "batch_router",
+		"destType": destType,
+	})
+
+	payloadSizeStat := stats.Default.NewTaggedStat("payload_size", stats.TimerType, map[string]string{
+		"module":   "batch_router",
+		"destType": destType,
+	})
+
+	startTime := time.Now()
+	payloadSizeStat.Observe(float64(len(payload)))
+	pkgLogger.Debugf("[Async Destination Maanger] File Upload Started for Dest Type %v", destType)
+	responseBody, statusCodeHTTP := misc.HTTPCallWithRetryWithTimeout(url, payload, config.GetDuration("HttpClient.marketoBulkUpload.timeout", 30, time.Second))
+	pkgLogger.Debugf("[Async Destination Maanger] File Upload Finished for Dest Type %v", destType)
+	uploadTimeStat.Since(startTime)
+	var bodyBytes []byte
+	var statusCode string
+	if statusCodeHTTP != 200 {
+		bodyBytes = []byte(`"error" : "HTTP Call to Transformer Returned Non 200"`)
+		return common.AsyncUploadOutput{
+			FailedJobIDs:  append(failedJobIDs, importingJobIDs...),
+			FailedReason:  string(bodyBytes),
+			FailedCount:   len(failedJobIDs) + len(importingJobIDs),
+			DestinationID: destinationID,
+		}
+	}
+
+	bodyBytes = responseBody
+	statusCode = gjson.GetBytes(bodyBytes, "statusCode").String()
+
+	var uploadResponse common.AsyncUploadOutput
+	switch statusCode {
+	case "200":
+		var responseStruct common.UploadStruct
+		err := json.Unmarshal(bodyBytes, &responseStruct)
+		if err != nil {
+			panic("Incorrect Response from Transformer: " + err.Error())
```

Comment:
Send a stat and fail the job. We can add an alert later.

Reply:
done